src/cpp/lpnamer/main/ProblemGeneration.cpp

```diff
@@ -425,16 +426,17 @@ void ProblemGeneration::RunProblemGeneration(
         reader->Delete();
     }
     FileWriter file_writer(configuration_manager_.Format());
-    MasterGeneration master_generation(xpansion_output_dir,
-                                       links,
-                                       additionalConstraints,
-                                       couplings,
-                                       master_formulation,
-                                       solver_config_.Name(),
-                                       logger,
-                                       solver_log_manager,
-                                       file_writer,
-                                       configuration_manager_.Format());
+    auto&& candidates = MasterGeneration(xpansion_output_dir,
+                                         solver_config_.Name(),
+                                         logger,
+                                         solver_log_manager,
+                                         file_writer,
+                                         configuration_manager_.Format())(links,
+                                                                          master_formulation,
+                                                                          additionalConstraints);
```

Comment:
Split in two for clarity
```suggestion
MasterGeneration environment(xpansion_output_dir,
                             solver_config_.Name(),
                             logger,
                             solver_log_manager,
                             file_writer,
                             configuration_manager_.Format());
auto&& candidates = environment(links, master_formulation, additionalConstraints);
```